pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -218,6 +221,55 @@ func (r *MigPlan) ListMigrations(client k8sclient.Client) ([]*MigMigration, erro
 	return list, nil
 }
 
+// ListTemplatePods - get list of pod templates, associated with a plan resource
+func (r *MigPlan) ListTemplatePods(client k8sclient.Client) ([]corev1.Pod, error) {
+	pods := []corev1.Pod{}
+	for _, template := range DefaultTemplates {
+		// Get resources
+		list := unstructured.UnstructuredList{}
```

Comment:
Listing resources with templates using the _unstructured_ api seems overly complicated and contrary to approach used for other _like_ processing.  Given this is a well known set of resources kinds, it seems more consistent and straight forward to just list them specifically.

Reply:
I'm guessing this is a holdover from the original implementation which allowed for arbitrary user-specified resources on the migplan (which is no longer here). In any case, we have a list of 8 different resource types, but the template path is different for each of them. So when you say "list them specifically", do you mean something along the line of instead of the current slice of:
```
type TemplateResource struct {
￼	Resource     string `json:"resource"`
￼	TemplatePath string `json:"templatePath"`
￼}
item0 = TemplateResource{
		Resource:     "deployment.apps",
		TemplatePath: ".spec.template",
	}
```
we'd have something like this:
```
item0 = TemplateResource{
		Resource:    ocappsv1.DeploymentConfig{},
		TemplatePath: ".spec.template",
}
```
(and whatever struct definition would match the explicit resource object).
Would that work with the fact that the template path is different for each? We can't just say resource.Spec.Template since in some cases it would be resource.spec.JobTemplate.Spec.Template.

Reply:
I was wondering why not just list Deployments etc and inspect the template part of the spec in a structured way to build the list of pods.